### PR TITLE
Added property in test model

### PR DESCRIPTION
**Pull request by Suparna-Acharya**
**Fixed Issues**

#21967
**Files Changed**

TestController.cs-Added application user id to createTestAsync method
TestAc.cs-Added createByUserId property
Test.cs-Added createByUserId property
ITestrepository.cs-Added useid as parameter to createTestAsync method
TestRepository.cs-Added useid as parameter to createTestAsync method
TestRepositoryTest.cs-Added ApplicationUser creation before every createTestAsync method
TestConductRepositoryTest,cs-Changed testinstruction from object to var

**Checks**

- [x] Naming Conventions
- [x] Server side code comments (proper English, grammar and no spelling mistakes)
- [ ] Client side code comments (proper English, grammar and no spelling mistakes)
- [x] Unused variables, methods, blank spaces and name spaces. 
- [x] Optimal Code and code formatting
- [x] Commit History is proper, no merge is done. 
- [x] Commit/pull request messages should be proper to justify your feature
- [x] All test cases are executed provided by tester.

**Known Issues**

N/A

**Comments**

N/A

---
This change is [Reviewable](https://reviewable.io/reviews/promact/trappist/338)

### CODE REVIEW

File: Trappist/src/Promact.Trappist.Test/Tests/TestsRepositoryTest.cs [hunk 1]
```diff
@@ -375,7 +415,11 @@ public TestsRepositoryTest(Bootstrap bootstrap) : base(bootstrap)
             var categoryAcList = Mapper.Map<List<DomainModel.Models.Category.Category>, List<CategoryAC>>(categoryList);
             //Creating Test
             var test = CreateTest("Maths");
-            await _testRepository.CreateTestAsync(test);
+            string userName = "suparna@promactinfo.com";
```
Comment (chintans) [inline Trappist/src/Promact.Trappist.Test/Tests/TestsRepositoryTest.cs]: MAJOR Remove this unused 'userName' local variable. [rule](http://server12:9005/coding_rules#rule_key=csharpsquid%3AS1481)

File: Trappist/src/Promact.Trappist.Test/Tests/TestsRepositoryTest.cs [hunk 2]
```diff
@@ -375,7 +415,11 @@ public TestsRepositoryTest(Bootstrap bootstrap) : base(bootstrap)
...
+            ApplicationUser user = new ApplicationUser() { Email = userName, UserName = userName };
```
Comment (chintans) [inline Trappist/src/Promact.Trappist.Test/Tests/TestsRepositoryTest.cs]: MAJOR Remove this unused 'user' local variable. [rule](http://server12:9005/coding_rules#rule_key=csharpsquid%3AS1481)

File: Trappist/src/Promact.Trappist.Test/Tests/TestsRepositoryTest.cs [hunk 3]
```diff
@@ -375,7 +415,11 @@ public TestsRepositoryTest(Bootstrap bootstrap) : base(bootstrap)
...
+            await _userManager.CreateAsync(user);
+            var applicationUser = await _userManager.FindByEmailAsync(user.Email);
```
Comment (chintans) [inline Trappist/src/Promact.Trappist.Test/Tests/TestsRepositoryTest.cs]: MAJOR Remove this unused 'applicationUser' local variable. [rule](http://server12:9005/coding_rules#rule_key=csharpsquid%3AS1481)

**Comment by chintans** [comment]
SonarQube analysis reported 3 issues
* ![MAJOR][MAJOR] 3 major

Watch the comments in this conversation to review them.

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'

File: Trappist/src/Promact.Trappist.DomainModel/Models/Test/Test.cs [hunk 4]
```diff
@@ -28,5 +29,7 @@ public class Test : BaseModel
         public virtual ICollection<TestAttendees> TestAttendees { get; set; }
         public virtual ICollection<TestQuestion> TestQuestion { get; set; }
         public virtual ICollection<TestCategory> TestCategory { get; set; }
+        public string CreatedByUserId { get; set; }
```
Comment (Krunal-Promact) [inline Trappist/src/Promact.Trappist.DomainModel/Models/Test/Test.cs]: foreign key attribute is missing.

File: Trappist/src/Promact.Trappist.Test/Tests/TestsRepositoryTest.cs [hunk 5]
```diff
@@ -340,7 +380,7 @@ public TestsRepositoryTest(Bootstrap bootstrap) : base(bootstrap)
             await _questionRepository.AddSingleMultipleAnswerQuestionAsync(questionAC, applicationUser.Id);
 
             var test = CreateTest("Maths");
-            await _testRepository.CreateTestAsync(test);
+            await _testRepository.CreateTestAsync(test,applicationUser.Id);
```
Comment (Krunal-Promact) [inline Trappist/src/Promact.Trappist.Test/Tests/TestsRepositoryTest.cs]: formatting is not applied here.

**Pull request merged**